cli/archive.go

```diff
@@ -0,0 +1,114 @@
+package cli
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/fs"
+	"math"
+	"os"
+	"path/filepath"
+
+	"go.viam.com/utils"
+	"golang.org/x/exp/maps"
+)
+
+// getArchiveFilePaths traverses the provided rootpaths recursively,
+// collecting the file paths of all regular files and returns them in a slice.
+// This list of paths should be passed to createArchive.
+func getArchiveFilePaths(rootpaths []string) ([]string, error) {
+	files := map[string]bool{}
+	for _, pathRoot := range rootpaths {
+		err := filepath.WalkDir(filepath.Clean(pathRoot), func(path string, info fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.Type().IsRegular() {
```

Comment:
might want to do one or both of:
- log here that you're excluding irregular file and say the path
- note in docs that symlinks and other special files are excluded

I have gotten bitten before by systems that ignore symlinks